source/developer/developer-setup.md

```diff
@@ -139,6 +139,59 @@ Any issues? Please let us know on our forums at: http://forum.mattermost.org
 
 Any issues? Please let us know on our forums at: http://forum.mattermost.org
 
+### Windows ###
+
+1. Install and setup Docker
+    1. If you are using Win10 Pro/Ent, you might want to use Docker for Windows
+        1. Install [Docker for Windows](https://docs.docker.com/docker-for-windows/)
+        2. Add the line `127.0.0.1 dockerhost` to `C:\Windows\System32\drivers\etc\hosts` using editor with administrator privilege
+    2. For other version or you prefer using VirtualBox, use Docker Toolbox
+        1. Install [Docker Toolbox](https://www.docker.com/products/docker-toolbox)
+        2. Run the `Docker Quickstart Terminal` and let it configure `default` machine
+        3. Get `default` IP by run `docker-machine ip default` in the terminal
+        4. Add the line `<Docker-IP> dockerhost` to `C:\Windows\System32\drivers\etc\hosts` using editor with administrator privilege
+2. Install and setup Node.Js
+    1. Download and install from [https://nodejs.org/](https://nodejs.org/)
+    2. Open `Node.js command prompt` and run `npm config set cache <short-path>` to set npm global cache to shorter path, the default path will likely cause path over 255 characters.
```

Comment:
Just checked the error, it's the `marked` in the dependencies causing this problem not the old nesting behavior. The setup was written with NodeJs 6.5.0 and npm 3.10.3. And I was fixed it with the cache path `cache=F:/npm-cache`

Here goes the error log:
```
{ webapp } patch-build » npm install                                                                                                                                                                          /f/repo/mattermost/src/github.com/mattermost/platform/webapp 2
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac: Cloning into 'F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac'...
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac: fatal: '/cygdrive/c/Users/Chikei/AppData/Roaming/npm-cache/_git-remotes/git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315/C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315' does not appear to be a git repository
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac: fatal: Could not read from remote repository.
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac:
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac: Please make sure you have the correct access rights
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac: and the repository exists.
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-b3263315 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-5bc8c347\69736482dbad685c398a5eec33a59b5ab06057ac:
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac: Cloning into 'F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac'...
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac: fatal: '/cygdrive/c/Users/Chikei/AppData/Roaming/npm-cache/_git-remotes/git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73/C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73' does not appear to be a git repository
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac: fatal: Could not read from remote repository.
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac:
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac: Please make sure you have the correct access rights
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac: and the repository exists.
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-https-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-56d2aa73 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-4e9f9ac2\69736482dbad685c398a5eec33a59b5ab06057ac:
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac: Cloning into 'F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac'...
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac: fatal: '/cygdrive/c/Users/Chikei/AppData/Roaming/npm-cache/_git-remotes/git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8/C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8' does not appear to be a git repository
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac: fatal: Could not read from remote repository.
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac:
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac: Please make sure you have the correct access rights
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac: and the repository exists.
npm ERR! git clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac:
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v6.5.0
npm ERR! npm  v3.10.3
npm ERR! code 128

npm ERR! Command failed: git -c core.longpaths=true clone C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8 F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac
npm ERR! Cloning into 'F:\Apps\babun\.babun\cygwin\tmp\npm-1284-39e1c9c9\git-cache-73fb0545\69736482dbad685c398a5eec33a59b5ab06057ac'...
npm ERR! fatal: '/cygdrive/c/Users/Chikei/AppData/Roaming/npm-cache/_git-remotes/git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8/C:\Users\Chikei\AppData\Roaming\npm-cache\_git-remotes\git-github-com-mattermost-marked-git-69736482dbad685c398a5eec33a59b5ab06057ac-60c6e7c8' does not appear to be a git repository
npm ERR! fatal: Could not read from remote repository.
npm ERR!
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR!
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:

```